ledger/artifactmanager/handler.go

```diff
@@ -815,13 +834,26 @@ func (h *MessageHandler) handleHotRecords(ctx context.Context, genericMsg core.P
 			inslog.Error(err)
 			continue
 		}
+
+		savedIndex, err := h.db.GetObjectIndex(ctx, &id, false)
+		if err != nil {
+			return nil, err
+		}
+
 		err = h.db.SetObjectIndex(ctx, &id, decodedIndex)
 		if err != nil {
 			inslog.Error(err)
 			continue
 		}
+
 		meta.TTL--
-		h.Recent.AddObjectWithTTL(id, meta.TTL)
+		h.Recent.AddObjectWithTLL(id, meta.TTL)
+		if savedIndex != nil {
+			err = h.Recent.MarkAsMine(id)
```

Comment:
No need to mark as mine here. When this handler is called, there are no records on my node.